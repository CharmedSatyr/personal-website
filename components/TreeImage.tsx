"use client";

import { useTheme } from "next-themes";

import Image from "components/Image";
import { Mode } from "constants/mode";
import { useEffect, useState } from "react";

const TreeImage = () => {
	const [mounted, setMounted] = useState(false);
	const { theme } = useTheme();

	useEffect(() => setMounted(true), []);

	if (!mounted) {
		return null;
	}

	return (
		<Image
			alt="tree"
			className="-mt-4 inline"
			src={`/static/images/${theme === Mode.dark ? "tree" : "tree-contrast"}.webp`}
			width={50}
			height={50}
		/>
	);
};

export default TreeImage;
