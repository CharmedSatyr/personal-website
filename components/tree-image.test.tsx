import { render } from "@testing-library/react";

import TreeImage from "./tree-image";

describe("TreeImage", () => {
	it("should render without crashing", () => {
		const { container } = render(<TreeImage />);

		expect(container).toBeInTheDocument();
	});

	it("should match the last snapshot", () => {
		const { container } = render(<TreeImage />);

		expect(container).toMatchSnapshot();
	});
});
