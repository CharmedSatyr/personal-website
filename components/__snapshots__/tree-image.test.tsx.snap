// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TreeImage should match the last snapshot 1`] = `
<div>
  <img
    alt="light tree"
    class="-mt-3 inline dark:hidden"
    data-nimg="1"
    decoding="async"
    height="40"
    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
    srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
    style="color: transparent;"
    width="40"
  />
  <img
    alt="dark tree"
    class="-mt-3 hidden dark:inline-block"
    data-nimg="1"
    decoding="async"
    height="40"
    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
    srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
    style="color: transparent;"
    width="40"
  />
</div>
`;
